### PR TITLE
finished app

### PR DESCRIPTION
**Pull request by iamnikitakostin**
Hi,

Just finished the app.
Please, be harsh on the review.
Also, not sure if I followed the best practices for the DB connection, as SQL connection seems to be different from SQLITE.
If not, please correct me, and I will re-do and resubmit.

Thank you!

### CODE REVIEW

**Review by chrisjamiecarter — CHANGES_REQUESTED** [review]
Hey @iamnikitakostin 👋,

Excellent work on your Flashcards project submission 🎉!

I have performed a peer review. Review/ignore any comments as you wish.

🟡 **Back options exits app**
In `Manage Stacks` and `Manage Flash Cards` selecting back exits the app instead of going back to the main menu. This may be intended behaviour? But is bad UX imo.

🟠 **Edit Options do not Work**
Selecting edit on Stack or Flashcard gives message saying updated successfully but without prompting for user input on what the update should be.

***

However, your project has the following issues 🔴 !
Please revisit to try understand and fix these issues 🔧 , as they relate to the projects requirements and they block my approval.

🔴 **Study with no Stacks**
Throws an exception.

🔴 **Stack names need to be Unique**
image

***

Feel free to reach out if you have any questions or if you'd like to collaborate further on any of these points 🆘 .

I will go ahead and wait for you to make your changes, commit and push (which will then notify me) 😊 !

Thanks,
@chrisjamiecarter 👍

File: Flashcards.iamnikitakostin/FlashCards/FlashCards/Data/DataContext.cs [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+        public DbSet<StudySession> StudySessions { get; set; }
+
+        protected override void OnConfiguring(DbContextOptionsBuilder optionsBuilder)
+        {
+            if (!optionsBuilder.IsConfigured)
+            {
+                optionsBuilder.UseSqlServer("Server=.;Database=Phonebook;Trusted_Connection=True;Encrypt=false;");
+            }
+        }
+        protected override void OnModelCreating(ModelBuilder modelBuilder)
```
[the remark this answers is not in the document]
Comment (iamnikitakostin) [reply Flashcards.iamnikitakostin/FlashCards/FlashCards/Data/DataContext.cs:14]: Hi Chris,

Thank you so much for the review!
I have fixed all the points you have outlined.
Could you please let me know if my app's structure is good?
I was trying to simulate the structure of a web-api app.

My apologies for submitting the app with that many errors initially.

Thank you,
Nikita

File: Flashcards.iamnikitakostin/FlashCards/FlashCards/Controllers/ConsoleController.cs [hunk 2]
```diff
@@ -1,89 +1,87 @@
 ﻿using Spectre.Console;
 
-namespace FlashCards.Controllers
+namespace FlashCards.Controllers;
```
Comment (chrisjamiecarter) [inline Flashcards.iamnikitakostin/FlashCards/FlashCards/Controllers/ConsoleController.cs:3]: 🟢 **File-Scoped Namespaces**

💡 Although, remember to leave a blank line between namespace and class

File: Flashcards.iamnikitakostin/FlashCards/FlashCards/Program.cs [hunk 3]
```diff
@@ -1,23 +1,39 @@
...
-namespace FlashCards
+namespace FlashCards;
+
+internal class Program : ConsoleController
 {
-    internal class Program
+    static void Main(string[] args)
     {
-        static void Main(string[] args)
+        IConfiguration config = new ConfigurationBuilder()
```
Comment (chrisjamiecarter) [inline Flashcards.iamnikitakostin/FlashCards/FlashCards/Program.cs:14]: 🟢 **Configuration Settings**

🥇

**Review by chrisjamiecarter — APPROVED** [review]
Hey @iamnikitakostin 👋,

Fantastic job getting your updates pushed through 💪 


⭐ Your project now fulfills all the requirements.

I will go ahead and mark as approved, keep up the excellent work on the next projects! 😊

Best regards,
@chrisjamiecarter 👍

**Pull request closed without merging**